webapp/src/views/Collection.vue

```diff
@@ -192,263 +216,216 @@
       }
     },
     created: function () {
-      this.fetchAssetList();
+      this.isBusy = true
+      this.retrieveAndFormatAsssets()
     },
     methods: {
```

Comment:
Everything below here has to do with refactoring the methods to resolve "callback hell" or "pyramid of doom".

I followed the rough principles from: http://callbackhell.com/

TL;DR:

1. Keep code shallow
2. Modularize
3. Don't nest functions. Give them names and place them at the top level of your program.
4. Create reusable functions.